tidy3d/plugins/smatrix/smatrix.py

```diff
@@ -92,32 +90,141 @@ class ComponentModeler(Tidy3dBaseModel):
         title="Folder Name",
         description="Name of the folder for the tasks on web.",
     )
-
-    batch: Tidy3dBaseModel = pd.Field(
+    element_mappings: Dict[Element, Dict[Element, Callable[[complex], complex]]] = pd.Field(
+        {},
+        title="Element Mappings",
+        description="Mapping between elements of the scattering matrix, as specified by"
+        "specified by pairs of ``(port name, mode index)`` matrix indices. "
+        "``element_mappings[element1][element2]`` returns a function ``f`` that sets the matrix "
+        "element2 as ``f(element1)``. "
+        "Each ``element`` is defined by a pair of indices ``((str, int), (str, int))`` relating "
+        "the source port name and mode index to the monitor port name and mode index.",
+    )
+    run_only: Optional[Tuple[MatrixIndex, ...]] = pd.Field(
+        None,
+        title="Run Only",
+        description="If specified, a tuple of matrix indices, specified by (:class:`.Port`, `int`),"
+        " to run only, excluding the other colulmns from the scattering matrix. "
+        "If this option is used, the resulting scattering matrix will not be square or complete.",
+    )
+    batch: Optional[Batch] = pd.Field(
         None,
         title="Batch",
-        description="Batch of task used to compute S matrix. Set internally.",
+        description="Batch containing all of the simulations needed for the scattering matrix." "",
```

Comment:
Shouldn't this say that it is set internally? I don't think there's a scenario where the user supplies it?

Reply:
I guess it's possible the user downloaded the batch separately from the component modeler and wants to supply it. But yea I'll mention to not use this unless you know what you're doing.